client/packages/host/src/components/Footer/UserDetails.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { FC } from 'react';
+import {
+  Box,
+  CircularProgress,
+  FlatButton,
+  PaperPopoverSection,
+  useAuthContext,
+  usePaperClickPopover,
+  useTranslation,
+  useNavigate,
+  useUserDetails,
+  useConfirmationModal,
+  RouteBuilder,
+  PowerIcon,
+  TextWithLabelRow,
+} from '@openmsupply-client/common';
+import { AppRoute } from '@openmsupply-client/config';
+import { PropsWithChildrenOnly } from '@common/types';
+
+export const UserDetails: FC<PropsWithChildrenOnly> = ({ children }) => {
+  const { logout, user, token } = useAuthContext();
+  const navigate = useNavigate();
+  const { hide, PaperClickPopover } = usePaperClickPopover();
+  const { isLoading } = useUserDetails(token);
+  const t = useTranslation();
+
+  const handleLogout = () => {
+    navigate(RouteBuilder.create(AppRoute.Login).build());
+    logout();
+  };
+
+  const showConfirmation = useConfirmationModal({
+    onConfirm: handleLogout,
+    message: t('messages.logout-confirm'),
+    title: t('heading.logout-confirm'),
+  });
+
+  const logoutButton = (
+    <FlatButton
+      startIcon={<PowerIcon fontSize="small" color="primary" />}
+      label={t('logout')}
+      disabled={false}
+      onClick={async () => {
+        hide();
+        showConfirmation();
+      }}
+      sx={{
+        whiteSpace: 'nowrap',
+        overflowX: 'hidden',
+        overflowY: 'visible',
+        textOverflow: 'ellipsis',
+      }}
+    />
+  );
+
+  return user ? (
+    <PaperClickPopover
+      placement="top"
+      width={300}
+      Content={
+        <PaperPopoverSection label={`${user.firstName} ${user.lastName}`}>
+          {isLoading ? (
+            <CircularProgress size={12} />
+          ) : (
+            <Box
+              sx={{
+                overflowY: 'auto',
+                overflowX: 'auto',
+                maxHeight: 300,
+                margin: '.5rem',
+              }}
+            >
+              <TextWithLabelRow
+                label={t('heading.username')}
+                text={user.name}
+                textProps={{
+                  textAlign: 'left',
+                  lineHeight: 1.5,
+                  textOverflow: 'ellipsis',
+                  whiteSpace: 'nowrap',
+                  overflow: 'hidden',
+                  width: 150,
+                }}
+                labelProps={{
+                  sx: { textAlign: 'left', width: 80, lineHeight: 1.5 },
+                }}
+                showToolTip={true}
+              />
+              <TextWithLabelRow
+                label={t('label.email')}
+                text={user.email ?? ''}
```

Comment:
```suggestion
                text={user.email ?? '-'}
```

Bit nicer to communicate no email set :) 

![Screenshot 2024-12-16 at 3 08 55 PM](https://github.com/user-attachments/assets/f4ef7b28-ad1f-44e9-9ff7-956230ff2cc3)

Reply:
global const UNDEFINED_STRING_VALUE 😉

Reply:
keeping me honest @roxy-dao  thank you 😁

Reply:
Updated here dbcdd53d553899815728e0849f76c11071c0009d :)